### PR TITLE
Upgrade Hugo to 0.52

### PR DESCRIPTION
**Pull request by lucperkins**
Hugo version 0.52 adds support for HTML minification via the `--minify` flag. This PR updates the "current" Hugo version to 0.52.

This change will _not_ impact users running the site locally with an earlier version of Hugo. This will *only* affect the production build of the site. For users who *do* upgrade to 0.52 to run the site locally, the [in-browser error](https://gohugo.io/news/0.50-relnotes/) feature is disabled to ensure a smooth transition. We can discuss enabling that (very nice but potentially disruptive) functionality in the future.

### CODE REVIEW

**Comment by zacharysarah** [comment]
@lucperkins Apart from netlify building successfully ([which it did](https://deploy-preview-11552--kubernetes-io-master-staging.netlify.com)) and travis passing ([which it did](https://travis-ci.org/kubernetes/website/builds/464065717?utm_source=github_status&utm_medium=notification)), is there a way we should be testing these changes prior to merge?

**Comment by lucperkins** [comment]
@zacharysarah I ran the site locally via both a standard Hugo binary and the Docker image and it worked just fine, as far as I can tell. I think that's about as much of a testing battery as we can throw at it. Perhaps it'd be useful to codify this somewhere?

**Comment by zacharysarah** [comment]
@lucperkins I didn't test with a docker image, but I also ran locally with success. What are you thinking we should codify outside the issue itself?

**Comment by zacharysarah** [comment]
@GuessWhoSamFoo Good point--I could foresee potential issues arising if [Dockerfile](https://github.com/kubernetes/website/blob/462b8ed2a25e83b30692af00b632288720c0b083/Dockerfile#L19) passed in `HUGO_VERSION` from a Makefile that doesn't match with `.travis.yml` and/or `netlify.toml`.

**Comment by chenopis** [comment]
This approach makes sense to me.
/LGTM

File: netlify.toml [hunk 1]
```diff
@@ -7,7 +7,7 @@ functions = "functions"
 command = "make non-production-build"
 
 [build.environment]
-HUGO_VERSION = "0.52"
+HUGO_VERSION = "0.49"
 
 [context.production.environment]
```
Comment (kbhawkey) [inline netlify.toml]: @lucperkins, can you further explain why the HUGO_VERSION in netlify.toml is remaining at 0.49? Who/what calls the non-production-build target (why should version 0.49 vs 0.52 matter)?

Comment (lucperkins) [reply netlify.toml]: Good catch. This should indeed also be 0.52. Fixed.

**Pull request merged**